proxy/src/telemetry/metrics/latency.rs

```diff
@@ -62,10 +63,11 @@ pub struct Histogram {
     /// Array of buckets in which to count latencies.
     ///
     /// The upper bound of a given bucket `i` is given in `BUCKET_BOUNDS[i]`.
-    buckets: [u32; NUM_BUCKETS],
+    // TODO: consider using `prometheus::Counter` rather than `Wrapping<u32>`?
+    buckets: [Wrapping<u32>; NUM_BUCKETS],
```

Comment:
It seems better to me to use `prometheus::Counter` here.

Reply:
It seemed better to me too; though note that the bucket counts are currently `u32`s while `prometheus::Counter` is a `u64`. Do you think doubling the size of the histogram data structure is worth the consistency in types?

Reply:
I understand we use `Counter` to count requests and responses and histograms for latencies. It seems reasonable to me to expect that the latencies for requests could at least sometimes be put into a few buckets (not spread out across all buckets) so it's not clear to me why the type for counting total requests is twice the size of the type for measuring latencies.

Put differently, my understanding is that NUM_BUCKETS = 26 so even assuming perfect distribution within the histogram, we're using `Counter` as 64-bits to handle the difference between supporting ~4 billion requests and ~104 billion requests before wrapping. But if wrapping is OK and we expect wrapping to be rare then why not just wrap all the counters at 4 billion increments, i.e. use a 32-bit type for them? Or if we think that's cutting things to close then is it really OK to use 32-bit counters for the histogram buckets?

I don't really have a strong opinion about it, but my point is that I don't understand the thinking behind the disparity in the first place.

Reply:
The histogram bucket counts were made 32-bits initially because this is how they were defined in the push metrics protobuf IDL, to reduce wire size.

I'd be happy to make them 64-bit values as well when the push metrics reporter is removed. I suppose I could also modify the push metrics reporter to accept 64-bit counts and cast them to `u32`, but I'd prefer to not touch that code at all, as it's slated to be removed very soon.